cpp/include/raft/linalg/detail/transpose.cuh

```diff
@@ -79,6 +80,55 @@ void transpose(math_t* inout, int n, cudaStream_t stream)
   });
 }
 
+template <typename T, typename LayoutPolicy>
+void transpose_row_major_impl(handle_t const& handle,
+                              device_matrix_view<T, LayoutPolicy> in,
```

Comment:
These should also be vanilla mdspan? I am surprised we are able to construct a `device_matrix_view` with mdspan

Reply:
I'm not sure if this should be Henry span. It's a safe guard that prevents users from passing the wrong pointer.

Reply:
It wouldn't be a breaking change when we support host mdspan since we will be just relaxing the constraint.

Reply:
I think we might be okay on the `impl` side. Because this isn't directly exposed to users, we have the freedom to change this as needed.

Reply:
`device_matrix_view` doesn't do any check on the pointer's underlying memory though, right? We can construct it even with a host mdspan

Reply:
Yeah I agree we should probably do a check there and raise exception for now if it's a host mdspan.

Reply:
I think that's why we should just accept vanilla mdspan all the way through

Reply:
It's a compile time guard. Given a host mdspan it won't compile.

Reply:
@trivialfis can we accept vanilla mdspan on our public API, and either use pointers or vanilla mdspan all the way after that? In the future, we will discuss the design of an "execution environment" that will allow us to handle memory guards better (compile time/run time). Until then, it will be easier to use the standard and then refactor if needed

Reply:
Got it, will make the changes.  Could you please loop me in on the idea of "execution environment"? I think that might benefit the development for XGBoost.

Reply:
Changed to using vanilla mdspan.